server/src/test/java/io/crate/planner/optimizer/symbol/CollectQueryCastRulesTest.java

```diff
@@ -170,7 +172,64 @@ public void test_any_operator_cast_on_right_reference_is_moved_to_cast_on_litera
     }
 
     @Test
-    public void test_any_operator_cast_on_nested_array_referewence_is_moved_to_cast_on_literal() {
+    public void test_all_operator_cast_on_left_reference_is_moved_to_cast_on_literal() {
+        for (var op : AllOperator.SUPPORTED_COMPARISONS) {
+            assertThat(toQuery("name " + op + " ALL([1, 2, 2])"))
+                .isFunction("_all_" + op,
+                    arg1 -> assertThat(arg1).isReference().hasName("name"),
+                    arg2 -> assertThat(arg2).isFunction("_cast")
+                );
+        }
+        for (var op : List.of("LIKE", "ILIKE")) {
+            String lowerOp = op.toLowerCase(Locale.ENGLISH);
+            assertThat(toQuery("name " + op + " ALL(d_array)"))
+                .isFunction("_all_" + lowerOp,
+                    arg1 -> assertThat(arg1).isReference().hasName("name"),
+                    arg2 -> assertThat(arg2).isFunction("_cast")
```

Comment:
This looks strange, the optimizer should move casts only to literals (as also the test name states), but here the cast is on a reference. Please clarify, and if valid these should move to a test which name describes what is tested.

Reply:
Thank you for pointing out, I ported `any` version of the tests without looking deeper into it. The tests for `any` was updated [here](https://github.com/crate/crate/commit/1a2b02017a008a1e321fadb4b8c96dbae4c657a5) but I think they actually should have been removed.

Reply:
What is happening with `<string ref> like any <double array ref>` is that since the only available `like any` function is `text, array(text)` variant, we end up casting the array(double) ref to array(text). And this has nothing to do with the optimizer rules.